numba/npyufunc/dufunc.py

```diff
@@ -52,6 +63,10 @@ def __init__(self, py_func, **kws):
             kws.pop('identity', None))
         super(DUFunc, self).__init__(dispatcher, **kws)
         self._install_type()
+        self._kernel = type('DUFuncKernel_' + self.ufunc.__name__,
```

Comment:
We've crossed an important threshold where we are discussing readability for a longer time than it would take to try to make improvements.  Patch committed.